utils.py

```diff
@@ -1,8 +1,95 @@
 import random
 import string
+import pandas as pd
+from google.cloud import bigquery
 
 random.seed(1236)
 
 # Generate a uuid of a specifed length(default=8)
 def generate_uuid(length: int = 8) -> str:
     return "".join(random.choices(string.ascii_lowercase + string.digits, k=length))
+
+
+def download_bq_table(bq_table_uri: str) -> pd.DataFrame:
+    """This function downloads a BigQuery table and returns a dataframe
+
+    Args:
+        bq_table_uri (str): BigQuery uri of the table
+
+    Returns:
+        pd.DataFrame: table data in dataframe format
+    """
+    # Remove bq:// prefix if present
+    prefix = "bq://"
+    if bq_table_uri.startswith(prefix):
+        bq_table_uri = bq_table_uri[len(prefix) :]
+
+    client = bigquery.Client()
+
+    table = client.get_table(bq_table_uri)
+    rows = client.list_rows(table)
+
+    return rows.to_dataframe()
+
+
+def save_dataframe_to_bigquery(dataframe: pd.DataFrame, table_name: str) -> str:
+    """This function loads a dataframe to a new bigquery table
+
+    Args:
+        dataframe (pd.Dataframe): dataframe to be loaded to bigquery
+        table_name (str): name of the bigquery table that is being created
+
+    Returns:
+        str: table id of the destination bigquery table
+    """
+    client = bigquery.Client()
```

Comment:
Yea, I guess we should be passing in one client instead of creating it multiple times. Let's see if it causes any issues...

Reply:
Yup! Makes a lot of sense :D